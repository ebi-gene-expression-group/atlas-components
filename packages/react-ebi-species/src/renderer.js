"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
