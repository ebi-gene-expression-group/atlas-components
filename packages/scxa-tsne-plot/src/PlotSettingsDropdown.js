import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'
import styled from 'styled-components'

// We replace the dropdown indicator for another component because the default chevron is a SVG element, not a
// background-image, whereas the control can be styled using the styles API packaged in React-Select

// Stolen from https://ebi.emblstatic.net/web_guidelines/EBI-Framework/v1.3/css/ebi-global.css select
const DropdownIndicator = styled.span`
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' version='1.1' width='32' height='24' viewBox='0 0 32 24'><polygon points='0,0 32,0 16,24' style='fill: rgb%28138, 138, 138%29'></polygon></svg>");
    background-origin: content-box;
    background-position: center;
    background-repeat: no-repeat;
    background-size: 9px 6px;
    width: 1.5rem;
    height: 1rem;
 `

const PlotSettingsDropdown = (props) => {
  const {labelText, options, onSelect, defaultValue} = props

  const formatGroupLabel = data => (
    <span>{data.label}</span>
  )

  const ebiVfSelectStyles = {
    control: (styles, state) => ({
      minHeight: `2.4375rem`,
      margin: `0 0 1rem`,
      padding: `0`,
      appearance: `none`,
      border: state.isFocused ? `1px solid #8a8a8a` : `1px solid #777`,
      borderRadius: `0`,
      backgroundColor: state.isDisabled ? `#e6e6e6` : `#fefefe`,
      fontFamily: `inherit`,
      fontSize: `1rem`,
      fontWeight: `normal`,
      lineHeight: `1.5`,
      color: `#0a0a0a`,
      transition: `box-shadow 0.5s, border-color 0.25s ease-in-out`,
      outline: state.isFocused ? `none` : `inherit`,
      boxShadow: state.isFocused ? `0 0 5px #777` : `none`,
      cursor: state.isDisabled ? `not-allowed` : `default`,
      display: `flex`
    }),
    menu: (styles) => ({
      ...styles,
      borderRadius: `0`,
      padding: `0`
    })
  }

  return [
    labelText && <label key={`label`}>{labelText}</label>,
    <Select
      key={`dropdown`}
      components={{ DropdownIndicator, IndicatorSeparator: null }}
      options={options}
      onChange={onSelect}
      defaultValue={defaultValue}
      formatGroupLabel={formatGroupLabel}
      styles={ebiVfSelectStyles} />
  ]
}

PlotSettingsDropdown.propTypes = {
  labelText: PropTypes.string,
  defaultValue: PropTypes.object,
  options: PropTypes.array.isRequired,
  onSelect: PropTypes.func.isRequired
}

export {PlotSettingsDropdown as default}

