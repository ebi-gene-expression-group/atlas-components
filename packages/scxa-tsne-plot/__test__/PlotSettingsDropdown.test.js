import React from 'react'
import renderer from 'react-test-renderer'

import {mount} from 'enzyme'

import Select from 'react-select'

import PlotSettingsDropdown from '../src/PlotSettingsDropdown'

describe(`PlotSettingsDropdown`, () => {
  test(`with no data matches snapshot`, () => {
    const onSelect = () => {}

    const tree = renderer
      .create(<PlotSettingsDropdown onSelect={onSelect} options={[]}/>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test(`contains Select component and label`, () => {
    const onSelect = () => {}

    const options = [
      {
        value: `hello`,
        label: `hello`
      },
      {
        value: `world`,
        label: `world`
      }
    ]

    const wrapper =
      mount(<PlotSettingsDropdown labelText={`Test dropdown`}
                                  defaultValue={
                                    {
                                      value: `world`,
                                      label: `world`
                                    }
                                  }
                                  options={options} onSelect={onSelect}/>)

    expect(wrapper.find(`label`).text()).toBe(`Test dropdown`)
    expect(wrapper.find(Select).length).toBe(1)
  })
})
