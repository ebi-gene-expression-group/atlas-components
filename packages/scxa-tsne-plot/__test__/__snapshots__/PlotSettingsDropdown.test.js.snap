// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlotSettingsDropdown with no data matches snapshot 1`] = `
<div
  className="css-10nd86i"
  onKeyDown={[Function]}
>
  <div
    className="css-12emboz"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="css-1hwfws3"
    >
      <div
        className="css-1492t68"
      >
        Select...
      </div>
      <div
        className="css-1g6gooi"
      >
        <div
          className=""
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            disabled={false}
            id="react-select-2-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "boxSizing": "content-box",
                "color": "inherit",
                "fontSize": "inherit",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "1px",
              }
            }
            tabIndex="0"
            type="text"
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1wy0on6"
    >
      <span
        className="sc-bdVaJa hUYIaS"
        cx={[Function]}
      />
    </div>
  </div>
</div>
`;
